src/sumBy.ts

```diff
@@ -1,4 +1,11 @@
 import { purry } from "./purry";
+import { type IterableContainer } from "./internal/types";
+
+type SumByBigInt<T, TS extends IterableContainer<T>> = TS extends []
+  ? 0
```

Comment:
This will not catch readonly arrays (e.g. `[] as const`)
```suggestion
type SumByBigInt<T, TS extends IterableContainer<T>> = TS extends readonly []
  ? 0
```

Reply:
Done